src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -179,6 +180,11 @@ public void RegisterWorkspaceHost(IVisualStudioWorkspaceHost host)
             _workspaceHosts.Add(new WorkspaceHostState(this, host));
         }
 
+        public Task RegisterWorkspaceHostFromForegroundThreadAsync(IVisualStudioWorkspaceHost host)
```

Comment:
I guess in general, if an API demands thread affinity, I'd expect the caller to deal with it, not add a new API for that particular caller. I'm not a fan of one-off methods for particular consumers.